Add format document for CCA attestation scheme (#51)
CCC-Attestation/attested-tls-poc: parsec-evidence-cca.md

# CCA Parsec Evidence

This document describes the format of the credential produced by the Parsec

> service as a key attestation credential for applications running in an Arm CCA realm.

Reviewer: ```suggestion
service as a key attestation credential for applications running in an Arm CCA Realm.
```

The Parsec service produces a blob of binary data as a response to a key
attestation operation issued by a client. The service takes in a nonce and uses
it as freshness for the attestation token.

The attestation token produced by Parsec is a CBOR encoded structure, namely a
[Combined Attestation Bundle
(CAB)](https://datatracker.ietf.org/doc/draft-bft-rats-kat/). The format is 
described below in CDDL.

```
parsecCcaAttestation = {
    "kat" => UCCS-KAT,

>     "pat" => CCA-token

Reviewer: For clarity sake and to avoid confusion I suggest we rename this to 
"ccat" => CCA-token

The reason been the name pat is strikingly similar to PAT (Platform Attestation Token) which has a different implications as per CCA Spec.

Author: Hmm, I think the reason for this naming is twofold - to be consistent with the TPM token, and to be consistent with the [KAT draft spec](https://www.ietf.org/archive/id/draft-bft-rats-kat-00.html#name-kat-pat-bundle). Are you refering here to the platform token section in the CCA attestation token? If so, I can try to make it clearer that this claim contains the whole token, not just that part.

Reviewer 2: I agree with Ionut that consistency is the driving factor here, even at the cost of some potential (but in any case very limited) confusion.

Reviewer: The confusion is also due to the following statement below, which also needs clarity:

`kat`: A sidecar token used to link the identity key of the application to the Platform Attestation Token (PAT) contained in `pat`. 

The above statement implies that there is PAT inside a `pat`

Reviewer: And by the way it is NOT a limited confusion but a serious confusion unless we clarify it adequately, and make is un-ambiguous to readers coming from CCA World!

Reviewer: @ionut-arm I understand your point on consistency with the TPM token and the specification.

We can resolve this with adequate clarity that:

The term `pat` and PAT referred in this document refers to entire CCA Token ( and CCA System) and is not tbe confused with the same term used in the CCA specification with a different scope.

Author: Yeah, agree, this seems to be down to the lack of clarity in my writing rather than anything else.. Will update the doc to improve on that.

}

UCCS-KAT = <TBD601>({
  &(eat_nonce: 10) => bstr .size (8..64)
  &(cnf: 8) => ik-pub
})

ik-pub = cnf-map

cnf-map = {
  &(cose-key: 1) => COSE_Key
}
```

- `pat`: An Arm CCA Attestation Token as defined in the [Arm CCA Realm Management Monitor specification](https://developer.arm.com/documentation/den0137/latest) (see "Attestation Token Format" section).

> - `kat`: A sidecar token used to link the identity key of the application to the Platform Attestation Token (PAT) contained in `pat`.

Reviewer: As mentioned above, the Line 34 creates the confusion and ambiguity.

Reviewer: We can re-word it as:

`kat`: A sidecar token used to link the identity key of the application to the Platform Attestation Token (PAT) identified by  `pat`.

- `eat_nonce`: A nonce issued by the challenger to prove the freshness of the attestation token.
- `cnf`: The public part of the application identity key, encoded as a [COSE_Key](https://www.rfc-editor.org/rfc/rfc9052#name-key-objects).

> The only signed object is the PAT, which is cryptographically linked to the Key Attestation Token (KAT)

Author: In this (current) doc I refer to the entire CCA attestation token as PAT (or `pat`) - as far as I can tell, the platform part of the CCA token is never refered to as `pat` in the specs.

[Combined Attestation Bundle
(CAB)](https://datatracker.ietf.org/doc/draft-bft-rats-kat/). The format is
described below in CDDL.

```
parsecCcaAttestation = {
    "kat" => UCCS-KAT,
    "pat" => CCA-token
}

UCCS-KAT = <TBD601>({
  &(eat_nonce: 10) => bstr .size (8..64)
  &(cnf: 8) => ik-pub
})

ik-pub = cnf-map

cnf-map = {
  &(cose-key: 1) => COSE_Key
}
```

- `pat`: An Arm CCA Attestation Token as defined in the [Arm CCA Realm
   Management Monitor
   specification](https://developer.arm.com/documentation/den0137/latest) (see
   "Attestation Token Format" section). Note that this token represents the
   entire credential issued by the Root of Trust, composed of a CCA platform
   token and a CCA Realm token. The Platform Attestation Token (PAT) referenced
   by this document should not be confused with the CCA platform token.
- `kat`: A sidecar token used to link the application identity key to the PAT.
- `eat_nonce`: A nonce issued by the challenger to prove the freshness of the
   attestation token.
- `cnf`: The public part of the application identity key, encoded as a
   [COSE_Key](https://www.rfc-editor.org/rfc/rfc9052#name-key-objects).

> The only signed object is the PAT, which is cryptographically linked to the Key

Reviewer: ```suggestion
The only signed object is the PAT, which is cryptographically linked to the Key
```

> The only signed object is the PAT, which is cryptographically linked to the Key

Reviewer: I would correct this statement as the PAT is not a signed object in itself but a collection itself which have signed entities.

Author: Oh, right, I forgot the two sections are signed separately.

> The only signed object is the PAT, which is cryptographically linked to the Key

Reviewer: ```suggestion
PAT contains collection of tokens, which are individually signed. PAT is cryptographically linked to the Key
```

Author: Done!
